test/evp_test.c

```diff
@@ -1572,6 +1574,28 @@ static int mac_test_run_mac(EVP_TEST *t)
             goto err;
         }
     }
+    if (reinit--) {
+        OSSL_PARAM ivparams[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+
+        /* If the MAC uses IV, we have to set it again */
+        if (expected->iv != NULL) {
+            ivparams[0] =
+                OSSL_PARAM_construct_octet_string(OSSL_MAC_PARAM_IV,
+                                                  expected->iv,
+                                                  expected->iv_len);
+            ivparams[1] = OSSL_PARAM_construct_end();
+        }
+        ERR_set_mark();
+        if (EVP_MAC_init(ctx, NULL, 0, ivparams)) {
+            ERR_clear_last_mark();
+            OPENSSL_free(got);
+            got = NULL;
+
+            goto retry;
+        }
+        /* If reinitialization fails, it is unsupported by the algorithm */
```

Comment:
Hmmm. This means the test is going to miss bugs where the initialisation fails when it should succeed. Perhaps we should have a list of algorithms that a re known to fail. If reinit fails for an algorithm that is expected to succeed then the test fails.

Reply:
Yeah, this is currently the Poly1305 only. I think I can add an option for that.

Reply:
Done